pkg/migrations/migrations.go

```diff
@@ -232,6 +236,50 @@ func migrationKey(migration migrationDescriptor) roachpb.Key {
 	return append(keys.MigrationPrefix, roachpb.RKey(migration.name)...)
 }
 
-func exampleNoopMigration(ctx context.Context, r runner) error {
+func checkQueryResults(results []sql.Result, numResults int) error {
+	if a, e := len(results), numResults; a != e {
+		return errors.Errorf("number of results %d != expected %d", a, e)
+	}
+	for _, result := range results {
+		if result.Err != nil {
+			return result.Err
+		}
+	}
 	return nil
 }
+
+// This migration is a little unusual in that it can only safely be run via a
+// stop-the-world upgrade. See #5887 for background discussion.
+func eventlogUniqueRowID(ctx context.Context, r runner) error {
+	// The series of statements to be run in order to update the system.eventlog
+	// table to replace "uniqueID BYTES DEFAULT experimental_unique_bytes()" with
+	// "uniqueID INT DEFAULT unique_rowid()". We have to truncate eventlog2 after
+	// creating it to avoid inserting duplicate rows in the case that it had been
+	// created but not renamed in a previous attempt.
+	modifySchemaStmts := []string{
+		fmt.Sprintf("CREATE TABLE IF NOT EXISTS system.eventlog2 %s;", sqlbase.NewEventLogTableFields),
+		"TRUNCATE TABLE system.eventlog2;",
+		"INSERT INTO system.eventlog2 SELECT timestamp, eventType, targetID, reportingID, info FROM system.eventlog;",
+		"DROP TABLE IF EXISTS system.eventlog;",
+		"ALTER TABLE system.eventlog2 RENAME TO system.eventlog;",
+	}
+
+	// If we don't disable event logging, dropping the eventlog table will fail
+	// when it tries to log the drop event.
+	sql.EventLoggingEnabled = false
```

Comment:
Do you think it's warranted? At this point there shouldn't be any concurrent access, but that's not to say that it won't start getting used for more things.